packages/hardhat-core/src/internal/hardhat-network/provider/transactions/ReadOnlyValidEIP1559Transaction.ts

```diff
@@ -68,7 +72,7 @@ export class ReadOnlyValidEIP1559Transaction extends FeeMarketEIP1559Transaction
       }
     );
 
-    super(data, { freeze: false, common: fakeCommon });
+    super(data, { ...opts, freeze: false, common: fakeCommon });
```

Comment:
This kind of change is needed to pass the `disableMaxInitCodeSizeCheck` flag around.

Reply:
See https://github.com/NomicFoundation/hardhat/pull/3767/files#r1143996647